src/main/scala/scalapb/json4s/JsonFormat.scala

```diff
@@ -723,11 +723,19 @@ class Parser private (config: Parser.ParserConfig) {
         res.fold[PValue](PEmpty)(PEnum.apply)
       }
       case ScalaType.Message(md) =>
-        fromJsonToPMessage(
-          containerCompanion.messageCompanionForFieldNumber(fd.number),
-          value,
-          false
-        )
+        try {
+          fromJsonToPMessage(
+            containerCompanion.messageCompanionForFieldNumber(fd.number),
+            value,
+            false
+          )
+        } catch {
+          case ex: JsonFormatException =>
+            throw new JsonFormatException(
+              s"Failed parsing field ${fd.name}: ${ex.getMessage}",
```

Comment:
With nested messages, wouldn't this lead to something that looks like:
`"Failed parsing field x: failed parsing field y: failed parsing field z: reason"`? Can this be changed somehow so the full path appears once?

Reply:
Hi @thesamet , sorry the person who opened that PR left my company so didn't follow up.
Indeed for this input `"""{"optMessage": { "optMessage": 39} }"""` the error message would be 
```
Failed parsing field opt_message: Failed parsing field opt_message: Expected an object for jsontest.MyTest, found JInt(39)
```
I don't know well this code base and I'm not seeing any trivial way to change it to a full path instead. I guess that might require subtyping JsonFormatException to accumulate a path for failure in nested objects, and I have no idea if adding that kind of complexity would be worth it or acceptable for the project.


Current implementation may not be the cleanest message, but it's simple and at least it's giving a helpful message, while without this implementation the message would be instead
```
Expected an object for jsontest.MyTest, found JInt(39)
```
Leaving the user clueless about where this is happening, especially for another type of error like "expected string and got number".